airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/connectors/test/argument_parsing.py

```diff
@@ -0,0 +1,67 @@
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+
+from __future__ import annotations
+
+import re
+from typing import TYPE_CHECKING
+
+import asyncclick as click
+
+if TYPE_CHECKING:
+    from typing import Callable, Dict, Tuple, Type
+    from enum import Enum
+    from pipelines.models.steps import STEP_PARAMS
+
+# Pattern  for extra param options: --<step_id>.<option_name>=<option_value>
+EXTRA_PARAM_PATTERN_FOR_OPTION = re.compile(r"^--([a-zA-Z_][a-zA-Z0-9_]*)\.([a-zA-Z_-][a-zA-Z0-9_-]*)=([^=]+)$")
+# Pattern  for extra param flag: --<step_id>.<option_name>
+EXTRA_PARAM_PATTERN_FOR_FLAG = re.compile(r"^--([a-zA-Z_][a-zA-Z0-9_]*)\.([a-zA-Z_-][a-zA-Z0-9_-]*)$")
+EXTRA_PARAM_PATTERN_ERROR_MESSAGE = "The extra flags must be structured as --<step_id>.<flag_name> for flags or --<step_id>.<option_name>=<option_value> for options. You can use - or -- for option/flag names."
+
+
+def build_extra_params_mapping(
+    SupportedStepId: Type[Enum]
```

Comment:
nit: we should pluralize this. (also I know this is my original mistake, sorry!)
```suggestion
    SupportedStepIds: Type[Enum]
```